gcc/rust/util/rust-attributes.cc

```diff
@@ -105,39 +109,77 @@ BuiltinAttributeMappings::BuiltinAttributeMappings ()
       mappings.insert ({def.name, def});
     }
 }
+class Attributes {
```

Comment:
There is already an `Attributes` class in `rust-attributes.h`, which you should add functions to.